src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacReorderValidator.java

```diff
@@ -0,0 +1,151 @@
+ * validator.
+ */
+public class IsaacReorderValidator implements IValidator {
+    private static final Logger log = LoggerFactory.getLogger(IsaacReorderValidator.class);
+
+    @Override
+    public final QuestionValidationResponse validateQuestionResponse(final Question question, final Choice answer) {
+        Validate.notNull(question);
+        Validate.notNull(answer);
+
+        if (!(question instanceof IsaacReorderQuestion)) {
+            throw new IllegalArgumentException(String.format(
+                    "This validator only works with IsaacReorderQuestion (%s is not ReorderQuestion)", question.getId()));
+        }
+
+        if (!(answer instanceof ItemChoice)) {
+            throw new IllegalArgumentException(String.format(
+                    "Expected ItemChoice for IsaacReorderQuestion: %s. Received (%s) ", question.getId(), answer.getClass()));
+        }
+
+        // These variables store the important features of the response we'll send.
+        Content feedback = null;                        // The feedback we send the user
+        boolean responseCorrect = false;                // Whether we're right or wrong
+
+        IsaacReorderQuestion reorderQuestion = (IsaacReorderQuestion) question;
+        ItemChoice submittedChoice = (ItemChoice) answer;
+
+        // STEP 0: Is it even possible to answer this question?
+
+        if (null == reorderQuestion.getChoices() || reorderQuestion.getChoices().isEmpty()) {
+            log.error("Question does not have any answers. " + question.getId() + " src: "
+                    + question.getCanonicalSourceFile());
+            feedback = new Content("This question does not have any correct answers");
+        }
+
+        if (null == reorderQuestion.getItems() || reorderQuestion.getItems().isEmpty()) {
+            log.error("ReorderQuestion does not have any items. " + question.getId() + " src: "
+                    + question.getCanonicalSourceFile());
+            feedback = new Content("This question does not have any items to choose from!");
+        }
+
+        // STEP 1: Did they provide a valid answer?
+
+        if (null == feedback && (null == submittedChoice.getItems() || submittedChoice.getItems().isEmpty())) {
+            feedback = new Content("You did not provide an answer");
+        }
+
+        Set<String> submittedItemIdSet = null;
+        Set<String> allowedItemIds;
+        if (null != submittedChoice.getItems() && null != reorderQuestion.getItems()) {
+            submittedItemIdSet = submittedChoice.getItems().stream().map(Item::getId).collect(Collectors.toSet());
+            allowedItemIds = reorderQuestion.getItems().stream().map(Item::getId).collect(Collectors.toSet());
+            if (!allowedItemIds.containsAll(submittedItemIdSet)) {
+                feedback = new Content("You did not provide a valid answer; it contained unrecognised items");
```

Comment:
Very minor, but could this have punctuation at the end? I'd go with `!` but a full stop is also okay.